# HG changeset patch
# User alta88@fixall.com
# Date 1711576212 -3600
# Parent  125968a5eedb406215dcf31f910839540117b325
Fix feed vandalization, Part I - folder/icon css
- lower opacity for folder name, not icon, when paused
- no opacity change for non feed folders
- increase visibility of error icon
- no !important (a poor css solution) for css "content"
- error, busy, new message indicators on the icon are css "content" (for screen readers);
  the icon url is not - it is "background-image"
- reenable error/busy "content" icons to overlap with new messages indicator
- no "fill" to override a site favicon
- for folders with multiple subscriptions, use a favicon (of first sub url found)

Authored by Alta88 and contributed to the Betterbird Project.
See https://github.com/Betterbird/thunderbird-patches/blob/main/LICENSE for license details.
All uses require attribution to the Author.

diff --git a/mail/base/content/widgets/folder-tree-row.mjs b/mail/base/content/widgets/folder-tree-row.mjs
--- a/mail/base/content/widgets/folder-tree-row.mjs
+++ b/mail/base/content/widgets/folder-tree-row.mjs
@@ -379,14 +379,13 @@ class FolderTreeRow extends HTMLLIElemen
   setFolderPropertiesFromFolder(folder) {
     if (folder.server.type != "rss") {
       return;
     }
-    const urls = !folder.isServer
-      ? lazy.FeedUtils.getFeedUrlsInFolder(folder)
-      : null;
-    if (urls?.length == 1) {
-      const url = urls[0];
-      this.icon.style = `content: url("page-icon:${url}"); background-image: none;`;
+    const urls = lazy.FeedUtils.getFeedUrlsInFolder(folder);
+    if (urls) {
+      this.dataset["isFeedFolder"] = "true";
+    } else {
+      delete this.dataset["isFeedFolder"];
     }
     const props = lazy.FeedUtils.getFolderProperties(folder);
     for (const property of ["hasError", "isBusy", "isPaused"]) {
       if (props.includes(property)) {
@@ -394,8 +393,14 @@ class FolderTreeRow extends HTMLLIElemen
       } else {
         delete this.dataset[property];
       }
     }
+    if (!urls || props.includes("hasError") || props.includes("isBusy")) {
+      this.icon.style = "";
+    } else {
+      let url = urls[0];
+      this.icon.style = `background-image: url("page-icon:${url}");`;
+    }
   }
 
   /**
    * Update this row's name label to match the new `prettyName` of the server.
diff --git a/mail/themes/shared/mail/about3Pane.css b/mail/themes/shared/mail/about3Pane.css
--- a/mail/themes/shared/mail/about3Pane.css
+++ b/mail/themes/shared/mail/about3Pane.css
@@ -375,13 +375,8 @@ body {
     &[data-server-type="rss"] > .container > .icon {
       background-image: var(--icon-rss);
     }
 
-    &[data-server-type="rss"] > ul > li:not([data-folder-type], [data-is-busy="true"]) .icon {
-      --icon-color: currentColor;
-      fill: var(--icon-color);
-    }
-
     &[data-folder-type="archive"] > .container > .icon {
       background-image: var(--icon-archive);
       --icon-color: var(--folder-color-archive);
     }
@@ -430,33 +425,35 @@ body {
       background-image: var(--icon-newsletter);
       --icon-color: var(--folder-color-newsletter);
     }
 
-    &[data-is-paused="true"] > .container > .icon,
-    &[data-is-paused="true"] > .container > .name,
-    &[data-is-paused="true"] ul > li > .container > .icon,
-    &[data-is-paused="true"] ul > li > .container > .name {
-      opacity: 0.6;
+    &[data-is-feed-folder] > .container > .icon {
+      background-image: var(--folder-pane-folder-rss);
+      background-size: 16px 16px;
     }
 
-    &[data-is-busy="true"] > .container > .icon {
-      content: var(--icon-clock) !important;
-      background-image: none;
-      --icon-color: var(--button-primary-background-color);
+    &[data-is-paused] > .container > .name,
+    &[data-server-type][data-is-paused] ul > li[data-is-feed-folder] > .container > .name {
+      opacity: 0.6;
+    }
+    &[data-is-busy] > .container > .icon,
+    &[data-has-error] > .container > .icon {
+       background-image: none !important;
+    }
 
-      @media (prefers-reduced-motion: no-preference) {
-        content: var(--icon-loading) !important;
-        animation: activity-indicator-3pane 1.05s steps(30) infinite;
-        object-fit: cover;
-        object-position: 0px 0;
+    &[data-is-busy] > .container > .icon::before {
+      content: url("chrome://global/skin/icons/loading.png");
+    }
+
+    @media (prefers-reduced-motion: reduce) {
+      &[data-is-busy] > .container > .icon::before {
+        content: var(--icon-clock);
       }
     }
 
-    &[data-has-error="true"] > .container > .icon {
-      content: var(--icon-warning) !important;
-      background-image: none;
-      fill: var(--color-amber-30);
-      stroke: var(--color-amber-60);
+    &[data-has-error] > .container > .icon::before {
+      content: var(--icon-warning);
+      fill: var(--color-red-60);
     }
 
     &[data-tag-key] > .container > .icon {
       background-image: var(--icon-tag);
